File: packages/shared/mongodb/PriceHistory.ts
import type { PriceSource } from "./types";

export interface PriceHistory {
  _id: string;
  purchase_id: string;
  platform: string;
  product_id: string;
  price_member: number | null;
  price_non_member: number | null;
  member_tier_required: string | null;
  currency: string;
  checked_at: string;
  source: PriceSource;
  evidence_screenshot_url: string | null;
  raw_response_hash: string | null;
}

File: packages/shared/mongodb/price_history.py
"""PriceHistory collection — mirror of PriceHistory.ts."""

from .base import BaseDocument
from .enums import PriceSource


class PriceHistory(BaseDocument):
    purchase_id: str
    platform: str
    product_id: str
    price_member: float | None
    price_non_member: float | None
    member_tier_required: str | None
    currency: str
    checked_at: str
    source: PriceSource
    evidence_screenshot_url: str | None
    raw_response_hash: str | None
